evm/contracts/examples/ConversionRate.sol

```diff
@@ -275,18 +276,36 @@ contract ConversionRate is ChainlinkClient, Ownable {
   }
 
   /**
-   * @dev Swaps the pointers to two uint256 arrays in memory;
+   * @dev Swaps the pointers to two uint256 arrays in memory
    * @param _a The pointer to the first in memory array
    * @param _b The pointer to the second in memory array
    */
-  function swap(uint256[] memory _a, uint256[] memory _b)
+  function swap(int256[] memory _a, int256[] memory _b)
     private
     pure
-    returns(uint256[] memory, uint256[] memory)
+    returns(int256[] memory, int256[] memory)
   {
     return (_b, _a);
   }
 
+  /**
+   * @dev Adds two int256s and makes sure the result doesn't overflow. Signed 
+   * integers aren't supported by the SafeMath library, thus this method
+   * @param _a The first number to be added
+   * @param _a The second number to be added
+   */
+  function safeAdd(int256 _a, int256 _b)
+    private
+    pure
+    returns (int256)
+  {
+    // solium-disable-next-line zeppelin/no-arithmetic-operations
+    int256 c = _a + _b; // signed integers are not supported by SafeMath
+    require(c >= _a, "SafeMath: addition overflow");
```

Comment:
This condition should not hold, if `_b` is negative. I think it's necessary to enumerate the cases for the possible signs of `_a` and `_b`, or something similar to that.

There should probably be tests for correct behavior and halting in the four cases, too.

Reply:
I suppose there are only two cases, actually: `_b` non-negative, and `_b` negative.